src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/data/MigrationContext.kt

```diff
@@ -1,10 +1,16 @@
 package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.data
 
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationType
 import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
 
-class MigrationContext<T>(val migrationId: String, val estimatedCount: Long, val body: T) {
-  constructor(context: MigrationContext<*>, body: T) : this(context.migrationId, context.estimatedCount, body)
```

Comment:
MigrationType is "the" enum representing what type of migration. Need so we can make the queue id generic